tests/by-util/test_pathchk.rs

```diff
@@ -164,3 +166,17 @@ fn test_posix_all() {
     // fail on empty path
     new_ucmd!().args(&["-p", "-P", ""]).fails().no_stdout();
 }
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_pathchk_non_utf8_paths() {
+    use uutests::at_and_ucmd;
+
+    let (at, mut ucmd) = at_and_ucmd!();
+    let filename = std::ffi::OsString::from_vec(vec![0xFF, 0xFE]);
+
+    // Create the file so pathchk can check it exists
+    std::fs::write(at.plus(&filename), b"test").unwrap();
+
+    ucmd.arg(&filename).succeeds();
```

Comment:
Creating a file is not necessary, `patchchk` operates only on file names.
```suggestion
    let filename = std::ffi::OsString::from_vec(vec![0xFF, 0xFE]);
    new_ucmd!().arg(&filename).succeeds();
```